packages/core/src/common/cluster/request-namespace-list-permissions.test.ts

```diff
@@ -0,0 +1,336 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type { V1SubjectRulesReviewStatus } from "@kubernetes/client-node";
+import type { DiContainer } from "@ogre-tools/injectable";
+import { getDiForUnitTesting } from "../../main/getDiForUnitTesting";
+import type { RequestNamespaceListPermissionsFor } from "./request-namespace-list-permissions.injectable";
+import requestNamespaceListPermissionsForInjectable from "./request-namespace-list-permissions.injectable";
+
+const createFakeProxyConfig = (statusResponse: Promise<{ body: { status: V1SubjectRulesReviewStatus }}>) => ({
+  makeApiClient: () => ({
+    createSelfSubjectRulesReview: (): Promise<{ body: { status: V1SubjectRulesReviewStatus }}> => statusResponse,
+  }),
+});
+
+describe("requestNamespaceListPermissions", () => {
+  let di: DiContainer;
+  let requestNamespaceListPermissions: RequestNamespaceListPermissionsFor;
+
+  beforeEach(() => {
+    di = getDiForUnitTesting({ doGeneralOverrides: true });
+    requestNamespaceListPermissions = di.inject(requestNamespaceListPermissionsForInjectable);
+  });
+
+  describe("when api returns incomplete data", () => {
+    it("returns truthy function", async () => {
+      const requestPermissions = requestNamespaceListPermissions(createFakeProxyConfig(
```

Comment:
If injection of dependencies were done using `injectables` instead of configuring curried functions manually, we could get rid of many steps of test initialisation here, namely everything but override of `createSelfSubjectRulesReviewInjectable`, (for which that would `asyncFn` fit nicely).

Reply:
Example:
```typescript
const createSelfSubjectRulesReviewMock = asyncFn();

di.override(createSelfSubjectRulesReviewInjectable, () => createSelfSubjectRulesReviewMock);

const requestPermissions = di.inject(requestPermissionsInjectable);
```

Reply:
I agree that the implementation could be simplified a lot but I'm not comfortable doing that in this PR.